Library/Homebrew/formulary.rb

```diff
@@ -638,17 +736,83 @@ def load_file(flags:, ignore_errors:)
       e.issues_url = tap.issues_url || tap.to_s
       raise
     end
+  end
 
-    private
+  # Loads a formula from a name, as long as it exists only in a single tap.
+  module FromNameLoader
+    sig {
+      params(ref: T.any(String, Pathname, URI::Generic), from: Symbol, warn: T::Boolean)
+        .returns(T.nilable(FromTapLoader))
+    }
+    def self.try_new(ref, from: T.unsafe(nil), warn: false)
+      return unless ref.is_a?(String)
+      return if ref.include?("/")
+
+      name = ref
+
+      loaders = Tap.map { |tap| FromTapLoader.try_new("#{tap}/#{name}") }
+                   .compact
+                   .select { _1.path.exist? }
+
+      case loaders.count
+      when 1
+        loaders.first
+      when 2..Float::INFINITY
+        # Always prefer the default tap, i.e. behave the same as if loading from the API.
+        if (default_tap_loader = loaders.find { _1.tap.core_tap? })
+          return default_tap_loader
+        end
```

Comment:
Is the issue fixed by #16654 relevant here?

Reply:
Yes. I will need to rebase this PR and apply the same fix here.

Reply:
Rebased and added a separate loader for the default tap.